test/skimNtuple2018_HHbtag.cpp

```diff
@@ -4273,168 +4276,71 @@ int main (int argc, char** argv)
-              try {kinFitsraw.fit();}
-              catch(HHKinFit2::HHInvMConstraintException   &e){wrongHHKraw=true;}
-              catch(HHKinFit2::HHEnergyConstraintException &e){wrongHHKraw=true;}
-              catch(HHKinFit2::HHEnergyRangeException      &e){wrongHHKraw=true;}
-              if(!wrongHHKraw)
-                {
-                  theSmallTree.m_HHKin_mass_raw             = kinFitsraw.getMH();
-                  theSmallTree.m_HHKin_mass_raw_chi2        = kinFitsraw.getChi2();
-                  theSmallTree.m_HHKin_mass_raw_convergence = kinFitsraw.getConvergence();
-                  theSmallTree.m_HHKin_mass_raw_prob        = kinFitsraw.getFitProb();
-                  if(DEBUG)
-                    {
-                       cout<<"---Kinfit Output debug---"<<endl;
-                       cout<<"KinFit mass = " << kinFitsraw.getMH() << endl;
-                       cout<<"KinFit chi2 = " << kinFitsraw.getChi2()<< endl;
-                       cout<<"KinFit conv = " << kinFitsraw.getConvergence()<< endl;
-                    }
-                }
-              else theSmallTree.m_HHKin_mass_raw = -100 ;
-              if ( (theBigTree.SVfitMass->at(chosenTauPair) > -900. || (doSmearing && theSmallTree.m_tauH_SVFIT_mass > 0)) && !wrongHHK)
+            // nominal kinfit raw
+            bool wrongHHKraw =false;
+            wrongHHKraw = fitterRaw.fit(whichSgnHp);
+            if(!wrongHHKraw)
               {
-                TLorentzVector b1 = kinFits.getFittedBJet1();
-                TLorentzVector b2 = kinFits.getFittedBJet2();
-                TLorentzVector bH_HKin = b1 + b2;
-                TLorentzVector tlv_HHsvfit = bH_HKin + tlv_tauH_SVFIT ;
-
-                theSmallTree.m_HHkinsvfit_bHmass = bH_HKin.M();
-                theSmallTree.m_HHkinsvfit_pt  = tlv_HHsvfit.Pt () ;
-                theSmallTree.m_HHkinsvfit_eta = tlv_HHsvfit.Eta () ;
-                theSmallTree.m_HHkinsvfit_phi = tlv_HHsvfit.Phi () ;
-                theSmallTree.m_HHkinsvfit_e   = tlv_HHsvfit.E () ;
-                theSmallTree.m_HHkinsvfit_m   = tlv_HHsvfit.M () ;
-              } // in case the SVFIT mass is calculated
-
-            } // end if doing HHKinFit
+                theSmallTree.m_HHKin_mass_raw             = fitterRaw.getMH();
+                theSmallTree.m_HHKin_mass_raw_chi2        = fitterRaw.getChi2();
+                theSmallTree.m_HHKin_mass_raw_convergence = fitterRaw.getConvergence();
+                theSmallTree.m_HHKin_mass_raw_prob        = fitterRaw.getFitProb();
+                if(DEBUG)
+                  {
+                     cout<<"---Kinfit Output debug---"<<endl;
+                     cout<<"KinFit mass = " << fitterRaw.getMH() << endl;
+                     cout<<"KinFit chi2 = " << fitterRaw.getChi2()<< endl;
+                     cout<<"KinFit conv = " << fitterRaw.getConvergence()<< endl;
+                  }
+              }
+            else theSmallTree.m_HHKin_mass_raw = -100 ;
```

Comment:
same comment here about the mass to be set to 0 in case of non-convergence

Reply:
Ok